plugins/search/providers/elastic_search/lib/model/items/ESearchCategoryEntryItem.php

```diff
@@ -0,0 +1,193 @@
+	);
+
+	private static $map_field_format = array(
+		ESearchCategoryEntryFieldName::ID => 'formatCategoryIdStatus',
+		ESearchCategoryEntryFieldName::NAME => 'formatCategoryNameStatus',
+		ESearchCategoryEntryFieldName::FULL_IDS => 'formatCategoryFullIdStatus',
+		ESearchCategoryEntryFieldName::ANCESTOR_ID => 'formatParentCategoryIdStatus',
+		ESearchCategoryEntryFieldName::ANCESTOR_NAME => 'formatParentCategoryNameStatus',
+	);
+
+	/**
+	 * @return ESearchCategoryEntryFieldName
+	 */
+	public function getFieldName()
+	{
+		return $this->fieldName;
+	}
+
+	/**
+	 * @param ESearchCategoryEntryFieldName $fieldName
+	 */
+	public function setFieldName($fieldName)
+	{
+		$this->fieldName = $fieldName;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getSearchTerm()
+	{
+		return $this->searchTerm;
+	}
+
+	/**
+	 * @param string $searchTerm
+	 */
+	public function setSearchTerm($searchTerm)
+	{
+		$this->searchTerm = $searchTerm;
+	}
+
+	/**
+	 * @return int
+	 */
+	public function getCategoryEntryStatus()
+	{
+		return $this->categoryEntryStatus;
+	}
+
+	/**
+	 * @param int $categoryEntryStatus
+	 */
+	public function setCategoryEntryStatus($categoryEntryStatus)
+	{
+		$this->categoryEntryStatus = $categoryEntryStatus;
+	}
+
+	public static function getAllowedSearchTypesForField()
+	{
+		return array_merge(self::$allowed_search_types_for_field, parent::getAllowedSearchTypesForField());
+	}
+
+	public static function createSearchQuery($eSearchItemsArr, $boolOperator, &$queryAttributes, $eSearchOperatorType = null)
+	{
+		$categoryEntryQuery = array();
+		$allowedSearchTypes = ESearchCategoryEntryItem::getAllowedSearchTypesForField();
+		$queryAttributes->setScopeToGlobal();
+		foreach ($eSearchItemsArr as $categoryEntrySearchItem)
+		{
+			self::getSingleItemSearchQuery($categoryEntrySearchItem, $categoryEntryQuery, $allowedSearchTypes, $queryAttributes);
+		}
+
+		return $categoryEntryQuery;
+	}
+
+	public static function getSingleItemSearchQuery($categoryEntrySearchItem, &$categoryEntryQuery, $allowedSearchTypes, &$queryAttributes)
+	{
+		$categoryEntrySearchItem->validateItemInput();
+		switch ($categoryEntrySearchItem->getItemType())
+		{
+			case ESearchItemType::EXACT_MATCH:
+				$categoryEntryQuery[] = self::getCategoryEntryExactMatchQuery($categoryEntrySearchItem, $allowedSearchTypes, $queryAttributes);
+				break;
+			case ESearchItemType::STARTS_WITH:
+				$categoryEntryQuery[] = self::getCategoryEntryPrefixQuery($categoryEntrySearchItem, $allowedSearchTypes, $queryAttributes);
+				break;
+			case ESearchItemType::EXISTS:
+				$categoryEntryQuery[] = self::getCategoryEntryExistsQuery($categoryEntrySearchItem, $allowedSearchTypes, $queryAttributes);
+				break;
+			default:
+				KalturaLog::log("Undefined item type[".$categoryEntrySearchItem->getItemType()."]");
+		}
+	}
+
+	protected static function getCategoryEntryExactMatchQuery($categoryEntrySearchItem, $allowedSearchTypes, &$queryAttributes)
+	{
```

Comment:
Create baseQueryCreateFactory
categoryEntryBaseFactory
Object type in factory for every type of search